main.rb

```diff
@@ -0,0 +1,24 @@
+def bubble_sort(arr)
+  n_times = arr.length - 1
+  switch = true
+  while switch
+    switch = false
+    n_times.times do |i|
+      arr[i], arr[i + 1] = arr[i + 1], arr[i] if arr[i] > arr[i + 1]
+      switch = true
+    end
+  end
```

Comment:
The `while` loop is running infinitely. This is because the terminating condition is never met, i.e., `switch` is always true. If the terminating condition is never met, the loop is going to run forever.

- [ ] Please investigate this and fix it.